### PR TITLE
poplar: misc updates

### PR DESCRIPTION
**Pull request by vchong**
LOAD_IMAGE_V2 support
OP-TEE header parsing
Misc updates

@shawnguo2 please test and ack. Thanks!

### CODE REVIEW

**Comment by shawnguo2** [comment]
I tested the patch set on Poplar board with different builds - Recovery vs. Normal, SPD=opteed vs SPD=none.  All work fine for me.

Tested-by: Shawn Guo <shawn.guo@linaro.org>

**Comment by vchong** [comment]
@shawnguo2 Great! Thanks!

Admins: Tags added. Please merge. Thanks!

**Comment by vchong** [comment]
Pushed an additional commit to fix minor typo in assignments for TSP_RAM_LOCATION variables.

File: plat/hisilicon/poplar/bl2_plat_setup.c [hunk 1]
```diff
@@ -9,11 +9,17 @@
 #include <bl_common.h>
 #include <console.h>
 #include <debug.h>
+#include <desc_image_load.h>
 #include <dw_mmc.h>
 #include <emmc.h>
 #include <errno.h>
 #include <generic_delay_timer.h>
 #include <mmio.h>
+#if LOAD_IMAGE_V2
```
Comment (davidcunado-arm) [inline plat/hisilicon/poplar/bl2_plat_setup.c]: No need to guard #includes - the compiler will optimise these out if unused.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@hzhuang1 
Please ACK?

**Comment by vchong** [comment]
@davidcunado-arm @hzhuang1 doesn't work on this board. @shawnguo2 is the lead for this project and has already acked above.

**Comment by davidcunado-arm** [comment]
Ok - no probs,

The maintainers list needs an update then - I'll try to get to it, but @shawnguo2 if you have a spare moment... :)

File: plat/hisilicon/poplar/bl2_plat_setup.c (hunk 1 above)
Comment (vchong) [reply plat/hisilicon/poplar/bl2_plat_setup.c]: ok, will remove

**Comment by vchong** [comment]
Thanks! I'll add another commit to update the maintainers list.

**Comment by vchong** [comment]
Updated

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vchong 
Could you split out the maintainers patch into its own PR, please?

**Comment by vchong** [comment]
@davidcunado-arm rebased and split maintainers patch (#1257).

**Comment by davidcunado-arm** [comment]
@vchong 
Thanks!

**Comment by davidcunado-arm** [comment]
@vchong 
Apologies - the optee_utils change - could you also move that commit to another PR; I don't want ACK from the optee maintainer to block yours, which looks good to merge now.

**Comment by vchong** [comment]
@davidcunado-arm np :) #1258

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
Check failures - issues in existing code that this PR is now including; can ignore.

**Pull request merged**